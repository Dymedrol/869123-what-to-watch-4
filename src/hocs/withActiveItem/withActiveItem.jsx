import React, {PureComponent} from "react";

const withActiveItem = (Component) => {
  class WithActiveItem extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        activeElement: null
      };

      this.setActiveItem = this.setActiveItem.bind(this);
      this.removeActiveItem = this.removeActiveItem.bind(this);
    }

    setActiveItem(item) {
      this.setState({
        activeElement: item
      });
    }

    removeActiveItem() {
      this.setState({
        activeElement: null
      });
    }

    render() {

      return (
        <Component
          {...this.props}
          setActiveItem={(item) => {
            this.setActiveItem(item);
          }}
          removeActiveItem={() => {
            this.removeActiveItem();
          }}
          activeElement = {this.state.activeElement}
        />
      );
    }
  }

  return WithActiveItem;
};

export default withActiveItem;
