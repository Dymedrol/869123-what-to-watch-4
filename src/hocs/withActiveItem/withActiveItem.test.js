import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";

import withActiveItem from "./withActiveItem.jsx";

describe(`Проверка снэпшота withActiveItem`, () => {
  const MockComponent = (props) => {
    const {children} = props;

    return (
      <div>
        {children}
      </div>
    );
  };

  MockComponent.propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
    ])
  };

  const MockComponentWrapped = withActiveItem(MockComponent);

  it(`withActiveItem is rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        setActiveItem = {false}
        onActiveItemChange = {() => {}}
        removeActiveItem = {() => {}}
      />), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
