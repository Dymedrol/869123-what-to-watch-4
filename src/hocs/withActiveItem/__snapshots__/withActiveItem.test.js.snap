// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка снэпшота withActiveItem withActiveItem is rendered correctly 1`] = `<div />`;
